spring-kafka/src/main/java/org/springframework/kafka/annotation/KafkaListenerAnnotationBeanPostProcessor.java

```diff
@@ -407,13 +407,19 @@ public Object postProcessAfterInitialization(final Object bean, final String bea
 					this.logger.debug(() -> annotatedMethods.size() + " @KafkaListener methods processed on bean '"
 							+ beanName + "': " + annotatedMethods);
 				}
+				Set<Method> methodsWithHandler = MethodIntrospector.selectMethods(targetClass,
+						(ReflectionUtils.MethodFilter) method ->
+								AnnotationUtils.findAnnotation(method, KafkaHandler.class) != null);
```

Comment:
I think no need to do this extraction outside of the `if (hasClassLevelListeners) {` branch.